Foundation/NSJSONSerialization.swift

```diff
@@ -97,7 +97,26 @@ public class NSJSONSerialization : NSObject {
     /* Generate JSON data from a Foundation object. If the object will not produce valid JSON then an exception will be thrown. Setting the NSJSONWritingPrettyPrinted option will generate JSON with whitespace designed to make the output more readable. If that option is not set, the most compact possible JSON will be generated. If an error occurs, the error parameter will be set and the return value will be nil. The resulting data is a encoded in UTF-8.
      */
     public class func dataWithJSONObject(obj: AnyObject, options opt: NSJSONWritingOptions) throws -> NSData {
-        NSUnimplemented()
+        guard obj is NSArray || obj is NSDictionary else {
```

Comment:
Right, that makes sense to tell a `String` from a `Int`, but iterating through all possible `[String: String]`, `[String: Int]`, [String: Double`, ... is less than ideal I think. I'm fine with using this for detecting different JSON-compatible types, but is it really the best way to detect different Array and Dictionary types?